tritony/tools.py

```diff
@@ -254,10 +254,27 @@ def _get_request_id(self):
         self.sent_count += 1
         return self.sent_count
 
-    def __call__(self, sequences_or_dict: Union[List[Any], Dict[str, List[Any]]]):
+    def __call__(
+        self,
+        sequences_or_dict: Union[List[Any], Dict[str, List[Any]]],
+        model_name: str = None,
+        model_version: str = None,
+    ):
         if self.triton_client is None:
             self._renew_triton_client()
 
+        if (model_name is not None and model_name != self.flag.model_name) or (
+            model_version is not None and model_version != self.flag.model_version
+        ):
+            if model_name is not None:
+                self.flag.model_name = model_name
+                if model_version is None:
+                    self.flag.model_version = "1"
+            if model_version is not None:
+                self.flag.model_version = model_version
+
+            self._renew_triton_client()
+
```

Comment:
local (optional) variable should not replace instance variable

```suggestion
current_model_name = model_name if model_name else self.model_name
current_model_version = model_version if model_version else self.model_version

```

Reply:
all codes below should be modified as above revision.